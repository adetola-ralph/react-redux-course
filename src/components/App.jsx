import React from 'react';
import { HashRouter as Router, Route, Switch } from 'react-router-dom';
import createBrowserHistory from 'history/createBrowserHistory';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

// Components
import Home from './home/HomePage';
import About from './about/AboutPage';
import Authors from './author/AuthorPage';
import Courses from './course/CoursesPage';
import ManageAuthor from './author/ManageAuthor';
import ManageCourse from './course/ManageCoursePage';
import EmptyList from './common/EmptyList';

import Header from './common/Header';

const customHistory = createBrowserHistory();

const NotFoundComponent = ({ history }) => {
  const goToHome = () => history.push('/');

  return (
    <EmptyList buttonAction={goToHome} buttonMessage="Click here to go Home" />
  );
};

NotFoundComponent.propTypes = {
  history: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
};

const App = ({ loading }) => (
  <Router history={customHistory}>
    <div className="container-fluid">
      <Header loading={loading} />
      <Switch>
        <Route exact path="/" component={Home} />
        <Route exact path="/about" component={About} />
        <Route exact path="/courses" component={Courses} />
        <Route exact path="/authors" component={Authors} />
        <Route exact path="/course/:id?" component={ManageCourse} />
        <Route exact path="/author/:id?" component={ManageAuthor} />
        <Route component={NotFoundComponent} />
      </Switch>
    </div>
  </Router>
);

App.propTypes = {
  loading: PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
  loading: state.ajaxCallsInProgress > 0,
});

export default connect(mapStateToProps)(App);
