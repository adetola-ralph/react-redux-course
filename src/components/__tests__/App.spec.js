import { create } from 'react-test-renderer';

import { App } from '../App';

describe('App component', () => {
  test('Snapshot', () => {
    const tree = create(<App loading />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
