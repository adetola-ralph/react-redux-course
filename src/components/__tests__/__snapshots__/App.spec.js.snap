// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Snapshot 1`] = `
<div
  className="container-fluid"
>
  <nav>
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    |
    <a
      href="/about"
      onClick={[Function]}
    >
      About
    </a>
    |
    <a
      href="/courses"
      onClick={[Function]}
    >
      Courses
    </a>
    |
    <a
      href="/authors"
      onClick={[Function]}
    >
      Authors
    </a>
    |
    <span>
      Number of courses: 
      0
    </span>
    <span>
      <span
        dots={20}
        interval={100}
      >
        .
         
      </span>
    </span>
  </nav>
  <div
    className="jumbotron"
  >
    <h1>
      Pluralsight Administration
    </h1>
    <p>
      React, Redux and React Router in ES6 for ultra responsive web apps.
    </p>
    <a
      className="btn btn-primary btn-lg"
      href="/about"
      onClick={[Function]}
    >
      Learn More
    </a>
  </div>
</div>
`;
